README.md

```diff
@@ -0,0 +1,115 @@
+# HBase Thirdparty
+<!---
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+-->
+
+This project packages relocated third-party libraries used by Apache HBase.
+
+> DISCLAIMER: This project is for Apache HBase internal use.  Included libs
+> and/or their versions are subject to change at the dictate of hbase without
+> regard to the concern of others!
+
+We have a number of submodules, one per ornery lib -- protobuf, netty, &c. --
+where we need special-handling and then a bucket for all the rest,
+hbase-shaded-miscellaneous. This latter includes protobuf-util, gson, and guava.
+
+General philosophy is many modules rather than a few fat ones so we can keep
+dependency narrow; a fat jar would put a load of unnecessaries on the
+CLASSPATH. The hbase-shaded-miscellaneous is a sort of all-the-rest but it
+is also libs that depend on each other and are awkward to disentangle.
+
+All shading is done using the same relocation offset of
+org.apache.hbase.thirdparty. We add this prefix to the relocated thirdparty
+library class names.
+
+See the pom.xml for the explicit version of each third-party lib included.
+
+Note that in hbase-shaded-protobuf, we unzip the protobuf jar to src/main/java
+rather than to a dir under target because the jar plugin wants src here (its
+hard to convince it otherwise). We also apply some patches. Current set are:
+
+```
+HBASE-15789_V2.patch
+HBASE-17087.patch
+HBASE-17239.patch
+```
+
+Ideally we would be pushing this set up into protobuf project.
+
+Note that this project requires JDK8. This is because a bunch of the code we
+have in hbase-unsafe is using old APIs that have been removed from more
+modern JDKs. Due to a bug in JDK, we cannot generate this code using a more
+modern version of the JDK. See
+[HBASE-26773](https://issues.apache.org/jira/browse/HBASE-26773) for details.
```

Comment:
Ah I hit https://issues.apache.org/jira/browse/HBASE-28380, while trying to build with JDK17. This doc helps me understand why we have hbase-unsafe module and how to release. Very useful. Thanks a ton :)